code/modules/events/wizard/magicarp.dm

```diff
@@ -51,7 +52,28 @@
 	color = "#00FFFF"
 	maxHealth = 75
 	health = 75
+	gold_core_spawnable = NO_SPAWN
 
 /mob/living/simple_animal/hostile/carp/ranged/chaos/Shoot()
 	projectiletype = pick(allowed_projectile_types)
 	..()
+
+/mob/living/simple_animal/hostile/carp/ranged/xenobiology // these are for the xenobio gold slime pool
+	gold_core_spawnable = HOSTILE_SPAWN
+	allowed_projectile_types = list(/obj/item/projectile/magic/change, /obj/item/projectile/magic/animate, /obj/item/projectile/magic/resurrection,
+	/obj/item/projectile/magic/teleport, /obj/item/projectile/magic/door, /obj/item/projectile/magic/aoe/fireball, /obj/item/projectile/magic/spellblade,
+	/obj/item/projectile/magic/arcane_barrage) //thanks Lett1
+
+/mob/living/simple_animal/hostile/carp/ranged/xenobiology/Initialize()
+	projectiletype = pick(allowed_projectile_types)
+	..()
+
+/mob/living/simple_animal/hostile/carp/ranged/chaos/xenobiology/Shoot()
+	projectiletype = pick(allowed_projectile_types)
+	..()
```

Comment:
67-73 can be removed.

Not only do vars inherit, but so do procs!

Reply:
Oh damn, didn't know procs also inherited, this one's gonna save me lots of space. Thanks for that one, I'll get it done in the next commit.